File: src/main/java/com/heybys/oddments/base/domain/LongTypeIdentifier.java
package com.heybys.oddments.base.domain;

public abstract class LongTypeIdentifier extends ValueObject<LongTypeIdentifier> {

    private final Long id;

    protected LongTypeIdentifier(Long id) {
        this.id = id;
    }

    public Long longValue() {
        return id;
    }

    public Long nextValue() {
        return id + 1;
    }

    @Override
    public boolean equals(Object other) {
        return super.equals(other);
    }

    @Override
    public int hashCode() {
        return super.hashCode();
    }

    @Override
    protected Object[] getEqualityFields() {
        return new Object[] {id};
    }

    @Override
    public String toString() {
        return this.getClass().getSimpleName() + " : " + longValue();
    }
}

File: src/main/kotlin/com/heybys/oddments/base/domain/LongTypeIdentifier.kt
package com.heybys.oddments.base.domain

abstract class LongTypeIdentifier(private val id: Long) : ValueObject<LongTypeIdentifier>() {

    fun longValue(): Long = id

    fun nextValue(): Long = id + 1

    override fun equals(other: Any?): Boolean {
        return super.equals(other)
    }

    override fun hashCode(): Int {
        return super.hashCode()
    }

    override val equalityFields: Array<Any?>
        get() = arrayOf(id)

    override fun toString(): String {
        return "${this.javaClass.simpleName} : $id"
    }
}
